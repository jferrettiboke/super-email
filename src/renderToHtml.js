import { renderToStaticMarkup } from "react-dom/server";
import inlineStyles from "./utils/inlineStyles";

const renderToHtml = element =>
  inlineStyles(`
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
${renderToStaticMarkup(element)}
`);

export default renderToHtml;
