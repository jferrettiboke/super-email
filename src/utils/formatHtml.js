import pretty from "pretty";

const formatHtml = html => pretty(html, { ocd: true });

export default formatHtml;
