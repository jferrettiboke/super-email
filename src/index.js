import renderToHtml from "./renderToHtml";
import Email from "./components/Email";
import Container from "./components/Container";
import Row from "./components/Row";
import Column from "./components/Column";
import Box from "./components/Box";
import Text from "./components/Text";
import Image from "./components/Image";
import Link from "./components/Link";

export {
  // Basic
  renderToHtml,
  Email,
  // Grid
  Container,
  Row,
  Column,
  // Other
  Box,
  Text,
  Image,
  Link
};
