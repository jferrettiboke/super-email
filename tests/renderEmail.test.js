import React from "react";

import { renderToHtml, Email } from "../src";
import formatHtml from "../src/utils/formatHtml";

test("renders email with title", () => {
  const html = renderToHtml(<Email title="Welcome">Welcome!</Email>);
  expect(formatHtml(html)).toMatchSnapshot();
});

test("renders email with preview", () => {
  const html = renderToHtml(
    <Email title="Welcome" preview="Confirm your email.">
      Welcome!
    </Email>
  );
  expect(formatHtml(html)).toMatchSnapshot();
});
