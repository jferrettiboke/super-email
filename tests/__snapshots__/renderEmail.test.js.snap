// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders email with preview 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\">
  <head>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\" />
    <title>Welcome</title>
    <style>
      * {
        margin: 0;
        padding: 0;
        border: none;
        box-sizing: border-box;
      }
    </style>
  </head>
  <body style=\\"margin:0;padding:0;box-sizing:border-box;background-color:white\\">
    <div style=\\"display:none;max-height:0;overflow:hidden\\">Confirm your email.</div>
    <div style=\\"display:none;max-height:0;overflow:hidden\\"> ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌
       ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌   ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ ‌ </div>
    <table data-component=\\"body\\" border=\\"0\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" style=\\"margin:0;padding:0;border:none;border-spacing:0;border-collapse:collapse;vertical-align:top;width:100%;height:100%;background-color:white\\">
      <tr style=\\"margin:0;padding:0;border:none;border-spacing:0;border-collapse:collapse;vertical-align:top;width:100%\\">
        <td align=\\"center\\" valign=\\"top\\" style=\\"margin:0;padding:0;border:none;border-spacing:0;border-collapse:collapse;vertical-align:top;width:100%;word-wrap:break-word;font-size:16px;line-height:1.3;color:#0a0a0a;font-family:&quot;Circular&quot;,Helvetica,Arial,sans-serif;font-weight:normal\\">
          <center style=\\"width:100%;min-width:580px\\">Welcome!</center>
        </td>
      </tr>
    </table>
  </body>
</html>"
`;

exports[`renders email with title 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Transitional//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\\">
<html xmlns=\\"http://www.w3.org/1999/xhtml\\">
  <head>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\" />
    <meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=UTF-8\\" />
    <title>Welcome</title>
    <style>
      * {
        margin: 0;
        padding: 0;
        border: none;
        box-sizing: border-box;
      }
    </style>
  </head>
  <body style=\\"margin:0;padding:0;box-sizing:border-box;background-color:white\\">
    <table data-component=\\"body\\" border=\\"0\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" style=\\"margin:0;padding:0;border:none;border-spacing:0;border-collapse:collapse;vertical-align:top;width:100%;height:100%;background-color:white\\">
      <tr style=\\"margin:0;padding:0;border:none;border-spacing:0;border-collapse:collapse;vertical-align:top;width:100%\\">
        <td align=\\"center\\" valign=\\"top\\" style=\\"margin:0;padding:0;border:none;border-spacing:0;border-collapse:collapse;vertical-align:top;width:100%;word-wrap:break-word;font-size:16px;line-height:1.3;color:#0a0a0a;font-family:&quot;Circular&quot;,Helvetica,Arial,sans-serif;font-weight:normal\\">
          <center style=\\"width:100%;min-width:580px\\">Welcome!</center>
        </td>
      </tr>
    </table>
  </body>
</html>"
`;
